html/mycakeapp/src/Controller/ReservesController.php

```diff
@@ -9,9 +9,194 @@ class ReservesController extends AppController
+            $seatReservation = $this->SeatReservations->find()
+                ->where([
+                    'member_id' => $_SESSION['seat']['member_id'],
+                    'schedule_id' => $_SESSION['seat']['schedule_id'],
+                    'column_number' => $_SESSION['seat']['column_number'],
+                    'record_number' => $_SESSION['seat']['record_number'],
+                ])
+                ->toArray();
+            $seatReservation[0]['is_cancelled'] = '1';
+            if (!($this->SeatReservations->save($seatReservation['0']))) {
+                return $this->redirect(['controller' => 'error']);
+            }
+            $this->request->session()->delete('seat');
+        }
+    }
+
+    public function ticket()
+    {
+        //座席予約から以下の4つの項目が渡されている前提(座席確保機能完成時に消す)
+        $_SESSION['seat']['member_id'] = 1;
+        $_SESSION['seat']['schedule_id'] = 1;
+        $_SESSION['seat']['column_number'] = 'A';
+        $_SESSION['seat']['record_number'] = 1;
+
+        //URL直打ち対策(途中でページを遷移したことによりセッションは残っていた場合も直接遷移させない)
+        if (empty($_SESSION['seat']) || $this->referer(null, true) !== '/reserves/seat' && $this->referer(null, true) !== '/reserves/ticket') {
+            return $this->redirect(['controller' => 'error']);
+        }
+        $this->viewBuilder()->setLayout('frame-title');
+        $title = 'チケット種別';
+        $schedule = $this->Schedules->get($_SESSION['seat']['schedule_id'], ['contain' => ['Movies']]);
+        $week = ['日', '月', '火', '水', '木', '金', '土'];
+        //映画情報及び座席情報をdetail変数に代入(viewへの呼び出す変数をまとめるため)
+        $detail['movieTitle'] = $schedule->movie->name;
+        $detail['date'] = $schedule->start_date->format('n月j日');
+        $detail['week'] = $week[$schedule->start_date->format('w')];
+        $detail['startTime'] = $schedule->start_date->format('G:i');
+        $detail['finishTime'] = date('G:i', strtotime('+' . $schedule->movie->screening_time . 'minute', strtotime($schedule->start_date)));
+        $detail['seatColumn'] = $_SESSION['seat']['column_number'];
+        $detail['seatRecord'] = $_SESSION['seat']['record_number'];
+        //基本料金情報を抽出
+        $tickets = $this->Fees->find()
+            ->where([
+                'started_at <=' => $schedule->start_date,
+                'OR' => [['finished_at >=' => $schedule->start_date], ['finished_at IS NULL']],
+                'is_deleted' => 0,
+            ])
+            ->order(['fee' => 'desc'])
+            ->toArray();
+        //取得した基本料金情報のID配列を作成(後々の照合用)
+        foreach ($tickets as $ticket) {
+            $ticketId[] = $ticket->id;
+        }
+        if ($this->request->is('post')) {
+            if (empty($this->request->data['detail'])) { //何も選択しない状態で送信した時
+                $notSelectError = '※チケット種別を選択してください';
+                $this->set(compact('notSelectError'));
+            }
+            $isTicketIdExist = in_array($this->request->data['detail'], $ticketId);
```

Comment:
選択しないで進むと$this->request->data['detail']が未定義のエラーが出てしまうので、else文の中に入れたらいいかなと思います！
```suggestion
            if (empty($this->request->data['detail'])) { //何も選択しない状態で送信した時
                $notSelectError = '※チケット種別を選択してください';
                $this->set(compact('notSelectError'));
            } else {
                $isTicketIdExist = in_array($this->request->data['detail'], $ticketId);
            }
```

Reply:
今までは大丈夫だったと思って確認不足でした💦
確認して修正いたします！！